src/main/java/core/basesyntax/service/impl/FruitServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitService;
+import core.basesyntax.service.strategy.impl.BalanceHandler;
+import core.basesyntax.service.strategy.impl.PurchaseHandler;
+import core.basesyntax.service.strategy.impl.ReturnHandler;
+import core.basesyntax.service.strategy.impl.SupplyHandler;
+import java.util.List;
+
+public class FruitServiceImpl implements FruitService {
+    @Override
+    public void fruitTransaction(List<FruitTransaction> fruitTransactionList) {
+        for (FruitTransaction transaction : fruitTransactionList) {
+            switch (transaction.getOperation().name()) {
+                case "BALANCE":
+                    new BalanceHandler().getOperation(transaction.getFruit());
+                    break;
+                case "SUPPLY":
+                    new SupplyHandler().getOperation(transaction.getFruit());
+                    break;
+                case "PURCHASE":
+                    new PurchaseHandler().getOperation(transaction.getFruit());
+                    break;
+                case "RETURN":
+                    new ReturnHandler().getOperation(transaction.getFruit());
+                    break;
+
+                default:
+                    throw new RuntimeException("Can`t be used operation !!");
+            }
```

Comment:
let's create Map<Operation, OperationHandler>, so you won't need such a switch case, pass this map as a parameter to the constructor